oxidation/libparsec/crates/client_types/src/local_device_file.rs

```diff
@@ -349,3 +358,74 @@ pub async fn load_recovery_device(
     LocalDevice::load(&plaintext)
         .map_err(|_| LocalDeviceError::Deserialization(key_file.to_path_buf()))
 }
+
+pub fn save_device_with_password(
+    key_file: &Path,
+    device: &LocalDevice,
+    password: &str,
+    force: bool,
+) -> Result<(), LocalDeviceError> {
+    // Safety check
+    assert_eq!(
+        key_file
+            .extension()
+            .unwrap_or_else(|| OsStr::new(""))
+            .to_str()
+            .unwrap(),
+        DEVICE_FILE_SUFFIX
+    );
```

Comment:
Not really a safety check ;) It just check the extension of a file.
In linux that won't identify the type of the file

Reply:
This assert was there in python. I removed it.